enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,956 @@
+HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value
+as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are res.hdr, ssl_c_der. Converters allowed are lower, base64."
+    Response []RouteHTTPHeader `json:"response,omitempty" protobuf:"bytes,1,opt,name=response"`
+    // request is a list of HTTP request headers to set or delete.
+    // The actions either Set or Delete will be performed on the headers in sequence as defined in the list of request headers.
+    // A maximum of 64 request header actions may be configured.
+    // You can use this field to specify HTTP request headers that should be set or deleted
+    // when forwarding connections from the client to your application.
+    // Note: This field cannot be used if your route uses TLS passthrough.
+    // + ---
+    // + Note: Any change to regex mentioned below must be reflected in the CRD validation of route in https://github.com/openshift/library-go/blob/master/pkg/route/validation/validation.go and vice-versa.    // +listType=map
+    // +listMapKey=name
+    // +optional
+    // +kubebuilder:validation:MaxItems=64
+    // +kubebuilder:validation:XValidation:rule=`self.all(key, key.action.type == "Delete" || (has(key.action.set) && key.action.set.value.matches('^(?:%(?:%|(?:\\{[-+]?[QXE](,[-+]?[QXE])*\\})?\\[(?:req\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]]*)$')))`,
+message="Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an
+HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value
+as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64."
+    Request []RouteHTTPHeader `json:"request,omitempty" protobuf:"bytes,2,opt,name=request"`
+}
+```
+
+```go
+// RouteHTTPHeader specifies configuration for setting or deleting an HTTP header.
+type RouteHTTPHeader struct {
+    // name specifies a header name to be set or deleted.  Its value must be a valid HTTP header.
+    // name as defined in RFC 2616 section 4.2.
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=256
+    // +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+    // +kubebuilder:validation:XValidation:rule="self.lowerAscii() != 'strict-transport-security'",message="strict-transport-security may not be set via header actions"
+    // +kubebuilder:validation:XValidation:rule="self.lowerAscii() != 'proxy'",message="proxy may not be set/delete via header actions"
```

Comment:
Is the `host` header allowed for Routes?

Reply:
yes